### DISCUSSION TITLE
Tough position for Stockfish

### DISCUSSION THREAD

**Discussion by douglaschess** [official-stockfish/Stockfish#4457] [General]
From a recent ICCF game. Stockfish 20230314 likes 44 .. f6? and can't see White is winning after 45 Nxf6! at depth of 86. Had to work backwards from finish to get it to see it.

[Event "WS/M/880"]
[Site "ICCF"]
[Date "2023.01.22"]
[Round "?"]
[White "Enev, Zlatko"]
[Black "Sazon, Adam"]
[Result "1-0"]
[ECO "B90"]
[WhiteElo "2201"]
[BlackElo "2365"]
[PlyCount "123"]
[EventDate "2023.??.??"]

1. e4 {https://www.iccf.com/game?id=1377735} c5 2. Nf3 d6 3. d4 cxd4 4. Nxd4 Nf6 5. Nc3 a6 6. Be3 e5 7. Nb3 Be6 8. Qd2 Nbd7 9. O-O-O Be7 10. h3 O-O 11. Kb1 b5 12. f3 Nb6 13. Qf2 Nc4 14. Bxc4 bxc4 15. Nc5 Qc7 16. N5a4 Rab8 17. Ka1 Rb7 18. g4 Rfb8 19. g5 Nh5 20. a3 Qc6 21. Qh4 g6 22. Qf2 Qc7 23. h4 Bf8 24. Rb1 Nf4 25. Ka2 Rb5 26. Ba7 R8b7 27. Be3 $1 Qc6 28. Rbd1 d5 29. Bxf4 $1 d4 30. Rxd4 $1 exd4 31. Qxd4 $1 Ra5 32. Qd8 $1 Rxa4 $1 33. Rd1 $1 Rd7 34. Rxd7 $1 Qxd7 $1 35. Qb6 $1 a5 36. Qc7 $2 $19 (36. Qe3 $142 $1 $13) 36... Qxc7 $1 37. Bxc7 Bd7 $1 38. Nd5 Bb5 39. Nf6+ Kh8 $17 (39... Kg7 40. Be5 c3 $1 41. Ne8+ Kg8 42. Nf6+ Kh8 43. Nd5+ Bg7 44. Bxg7+ Kxg7 $1 45. Nxc3 Rc4 $1 46. Nxb5 Rxc2 $1 47. Kb3 Rh2 $1 48. Ka4 Rxh4 $1 49. Nc7 f6 $1 50. Kxa5 fxg5 51. b4 Rh3 $1 52. b5 Rxf3 $1 $19) 40. c3 Kg7 41. Be5 Bc6 $13 (41... Be7 $17) 42. Nh5+ Kg8 43. Nf6+ $1 Kh8 44. Nd5+ f6 $2 $18 (44... Kg8 $142 $13) ({Stockfish dev-20230314-f0556dcb:} 44... f6 45. Nxf6 Kg7 46. Bd4 Be7 47. Ng4+ Kf8 48. Bf6 Bc5 49. Be5 Kf7 50. Nf6 Bf2 51. Nxh7 Bxh4 52. Bf4 Bf2 53. Nf6 Bc5 54. Ng4 Ke6 55. Kb1 Be8 56. Ka2 Be7 57. Ne3 Bd7 58. Nd5 Bc5 59. Nf6 Bc8 60. Bc1 Ba6 61. Bf4 Bb7 62. Bc1 Bc6 63. Bf4 Be7 64. Ng4 Bb5 65. Ne3 Bd8 66. Ng4 Kf7 67. Ne3 Kg7 68. Be5+ Kf8 69. Bf6 Bc7 70. Nd5 Bb8 71. Bd8 Bd6 72. Bc7 Bc5 73. Bf4 Kf7 74. Be3 Be7 75. Bf4 Be8 76. Ne3 Bd7 77. Nc2 Be6 78. Be3 Bd6 79. Bd4 Bd7 80. Be3 Be5 81. Bc1 Be6 82. Bd2 Bg7 83. Bc1 {-1.39/66}) 45. Nxf6 $1 ({Stockfish dev-20230314-f0556dcb:} 45. Nxf6 Kg7 46. Bd4 Be7 47. Ng4+ Kg8 48. Bf6 Bc5 49. Be5 Kf7 50. Nf6 Bf2 51. Nxh7 Bxh4 52. Bf4 Bf2 53. Nf6 Ke6 54. Ng4 Bb6 55. Nf6 Bd8 56. Ng4 Bb5 57. Ka1 Ke7 58. Ka2 Be8 59. Nf6 Bc6 60. Ng4 Ke6 61. Ne5 Be8 62. Ng4 Be7 63. Kb1 Bc5 64. Ka2 Ke7 65. Bh2 Bb6 66. Nf6 Bc6 67. Bf4 Ke6 68. Ng4 Bb5 69. Ne5 Be8 70. Ng4 Bc5 71. Nf6 Bf7 72. Ng4 Be7 73. Nh6 Be8 74. Ng4 Kf7 75. Kb1 Kg7 76. Be5+ Kf8 77. Bf6 Bd6 78. Be5 Bc5 79. Bf4 Bb6 80. Ka2 Kg7 81. Be5+ Kf7 82. Bf6 Ke6 83. Be5 Ba7 84. Bf4 Ke7 85. Bg3 Kd8 86. Bd6 Bb6 87. b3 cxb3+ 88. Kxb3 Ba7 {-1.17/86}) 45... Kg7 (45... Bb5 {Stockfish dev-20230314-f0556dcb:} 46. Bd4 Ba6 47. Nd7+ Kg8 48. Bc5 Bg7 49. Nb6 Bb5 50. b3 cxb3+ 51. Kxb3 Be2 52. Kxa4 Bxc3 53. f4 Kf7 54. Nd5 Be1 55. Be3 Bf3 56. Nf6 Bxh4 57. e5 Be1 58. Kb5 Bc3 59. Bc1 Ke6 ({Stockfish dev-20230314-f0556dcb:} 59... Ke6 60. Nxh7 Be4 61. Nf6 Bd3+ 62. Kc5 a4 63. Ne8 Be1 64. Kd4 Bb1 65. Nd6 Bc2 66. Be3 Ba5 67. Kc4 Bd8 68. Nb5 Bb3+ 69. Kb4 Be7+ 70. Bc5 Bd8 71. Nc3 Kf5 72. Nxa4 Be6 73. Be3 Be7+ 74. Nc5 Bd7 75. Kc4 Bc6 76. Kd4 Be8 77. Bc1 Bc6 78. Nd3 Ba4 79. Nb4 Bf8 80. Nd5 {3.32/32}) 60. Nxh7 Be2+ 61. Kc5 Be1 62. Be3 Bc3 63. Nf8+ Kf7 64. Nd7 Ke6 65. Kc6 Bf3+ 66. Kc7 Bb2 67. Nc5+ Ke7 68. a4 Bc3 69. Kb6 Kf7 70. Na6 Bd1 71. Nc7 Bb3 72. Kc6 Bb4 73. Kb5 Be1 74. Bc5 Bd2 75. Bd6 Bc3 76. Ka6 Bxa4 {1.74/53}) 46. Nd7+ $1 Kf7 47. Nxf8 $1 Kxf8 48. Kb1 $1 Bd7 49. Kc2 Ke8 50. Kd2 $1 Bc8 51. h5 gxh5 52. Bg3 Kf7 53. Ke3 h4 54. Bxh4 $1 Kg6 55. f4 Bd7 56. Kd4 Kh5 57. Be1 Kg4 58. Bd2 Kf3 59. f5 Ke2 60. Be3 Bc8 61. Ke5 Kxe3 62. f6 1-0

**Comment by RogerThiede** [comment]
What is the FEN of the tough position?

**Reply by dav1312** [reply]
`5b1k/5p1p/2b3p1/p2NB1P1/r1p1P2P/P1P2P2/KP6/8 b - - 8 44`

**Reply by RogerThiede** [reply]
Is there criteria for a _tough position_?  This position has only 3 legal moves, f6, Kg8, and Bg7.
44... f6 is not only not a mistake, it's the only move that maintains the advantage after searching with all tablebases.

`info depth 57 seldepth 60 multipv 1 score cp 145 wdl 916 84 0 nodes 2583700696 nps 847297 hashfull 10 tbhits 3117840 time 3049343 pv f7f6 d5f6 h8g7 e5d4 f8e7 f6g4 g7f8 d4f6 e7c5 f6e5 f8f7 g4f6 c5f2 f6h7 f2h4 e5f4 h4f2 h7f6 f2c5 f6g4 f7e7 g4e5 c6e8 e5g4 e7d8 g4f6 e8c6 f6g4 c5f8 g4f6 f8g7 f6g4 d8e7 g4e3 g7f8 e3c2 c6d7 f4e3 e7e8 e3c1 f8c5 c1d2 c5d6 d2c1 d7e6 c2d4 e6g8 d4c2 g8f7 c1e3 e8d7 c2d4 f7e8 d4c2 d7c8 c2d4 d6c5 d4c2 c5e7 e3d2`
`info depth 57 seldepth 27 multipv 2 score cp 0 wdl 4 992 4 nodes 2583700696 nps 847297 hashfull 10 tbhits 3117840 time 3049343 pv h8g8 d5f6 g8h8 f6g4 f7f6 e5d4 c6b5 f3f4 h8g7 f4f5 g6f5 e4f5 b5d7 h4h5 g7h8 g4f6 d7f5 h5h6 f5e6 d4e5 e6h3 e5d4 h3f5 d4e5`
`info depth 57 seldepth 40 multipv 3 score cp 0 wdl 4 992 4 nodes 2583700696 nps 847297 hashfull 10 tbhits 3117840 time 3049343 pv f8g7 e5d6 h7h6 d5e7 c6b5 e7d5 b5c6`

Quiescent search at depth 1 immediately recommended f6 as well.

**Reply by Craftyawesome** [reply]
No, OP is right. Backwards analyze the main line and also ...Bg7 Bxg7. Kg8 is the only drawing move, the other two lose for black.
2pv search:
![image](https://user-images.githubusercontent.com/19976035/228166108-8110c33e-c635-42a2-b45b-5329de246a41.png)

**Reply by douglaschess** [reply]
Without an understanding of how Stockfish really works, it seems to me like it's pruning out the best line and never resurrecting it no matter how deep you run it. Not sure if people consider this a an issue or a feature. Obviously the result is not ideal.

**Reply by PGG106** [reply]
"Not sure if people consider this a an issue or a feature", it's neither, it's a byproduct of the fact a chess engine has to prune nodes to even exist, single "tough" positions are not relevant, the result isn't ideal because in an ideal world we solved chess ( we didn't)

**Reply by Craftyawesome** [reply]
Isn't the rook just trapped? Unless there is a forcing line that wins it quickly and keeps winning advantage it will be hard to eval correctly. Either NNUE needs to realize how garbage the rook is or every single try by black needs to be searched deeply enough to fall apart. That's just too many positions.
